src/test/java/org/opensearch/ad/transport/ADBatchAnomalyResultTransportActionTests.java

```diff
@@ -137,7 +137,7 @@ public void testHistoricalDetectorWithValidDateRange() throws IOException, Inter
         DetectionDateRange dateRange = new DetectionDateRange(startTime, endTime);
         ADBatchAnomalyResultRequest request = adBatchAnomalyResultRequest(dateRange);
         client().execute(ADBatchAnomalyResultAction.INSTANCE, request).actionGet(5000);
-        Thread.sleep(10000);
+        Thread.sleep(20000);
```

Comment:
This will make our tests slower and slower when adding more tests.  We want to avoid even though the execute finishes quickly and we still sleep 20 seconds.  You can use a CountDownLatch to wait while the client().execute is running.  See an example in EntityProfileRunnerTests.stateTestTemplate

Reply:
Thank you very much for the suggestion! 
I have checked with Yaliang, he is still working on the batching change and need more time to optimize the unit tests.
I'm planning to update these unit tests after Yaliang finished his PRs.

Reply:
I will fix UTs related with AD batch tasks after all changes ready. I'm ok if @spbjss willing to put some effort to fix it now or leave it to later to avoid duplicate effort.

Reply:
sounds good. Will approve the PR once you get the build issue figured out.  Please let me know.